lib/cartopy/io/img_tiles.py

```diff
@@ -602,20 +619,26 @@ def __init__(self, access_token, username, map_id,
             tiles will be retrieved through this process. Note that this style
             may be private and if your access token does not have permissions
             to view this style, then map tile retrieval will fail.
-        desired_tile_form: optional
-            The desired tile format. Use 'RGBA' if desired style includes
-            transparency.
-
+        desired_tile_form : str, optional
+            The desired format of the tile (defaults to "RGB").
+        cache : bool or pathlib.Path or str, optional
+            To allow offline use, as well as not spam tile providers, Cartopy
+            can create a local cache of previously fetched tiles. The default
+            path is ``cartopy.config["cache_dir"]``. If it is set to ``True``,
+            the default path is used. If it is set to a custom path, then this
+            path is used instead of the default one. If it is set to ``False``,
+            the tiles are downloaded each time.
         """
+
         self.access_token = access_token
         self.username = username
-        self.map_id = map_id
```

Comment:
This is still breaking because someone could have been setting `my_mapbox_provider.map_id = my_new_style`.

```python
@property
def map_id(self):
    return self.style

@property.setter
def map_id(self, val):
    self.style = val
```